src/Compute/Compute/VirtualMachine/Operation/UpdateAzureVMCommand.cs

```diff
@@ -223,7 +228,9 @@ public override void ExecuteCmdlet()
                         Host = this.IsParameterBound(c => c.HostId)
                              ? new SubResource(this.HostId)
                              : this.VM.Host,
-                        VirtualMachineScaleSet = this.VM.VirtualMachineScaleSet,
+                        VirtualMachineScaleSet = this.IsParameterBound(c => c.VirtualMachineScaleSetId)
+                                                ? new SubResource(this.VirtualMachineScaleSetId)
```

Comment:
what happens if -VirtualMachineScaleSetId and -VM are both provided?  
Then here in this line -VM gets overwritten. 

It should be using -VM and only the vm.virtualmachinescaleset.id part gets overwritten, right?

Reply:
The property VM.VirtualMachineScaleSet is  a subresource. So when user want's to attach to a new VMSS by providing VirtualmachinescalesetId parameter it should overwrite the VM.VirtualMachineScaleSet property.

Reply:
I see. the only property of subresource is ID, so makes sense.

Reply:
that is confusing, but it is true. Just the naming is odd.

Reply:
The naming is is that way because it's a subresource.